src/wordladder.cpp

```diff
@@ -1,116 +1,307 @@
+    cout << "Welcome to CS 106B/X Word Ladder!" << endl;
+    cout << "Please give me two English words, and I will convert the" << endl;
+    cout << "first into the second by modifying one letter at a time" << endl;
+    cout << endl;
 
     //Ask for the dictionary file name
     getDictionary(dictionary);
+    cout << endl;
 
-    getWords(dictionary);
-
-    //Output word ladder "A ladder from [word 2] back to [word1]: [word ladder]
-    getWordLadder(dictionary);
-
-    //Repeat the sequence
+    //Repeat the following sequence
+    while (true) {
+        // Ask for word 1 and quit the program if it is blank
+        string wordOne = getWord("Word 1 (or Enter to quit): ");
+        if (emptyWord(wordOne)) {
+            break;
+        }
+        // Ask for word 2 and quit the program if it is blank
+        string wordTwo = getWord("Word 2 (or Enter to quit): ");
+        if (emptyWord(wordTwo)) {
+            break;
+        }
 
+        // Perform validation checks on word 1 and word 2 before proceeeding to word ladder stage
+        if (validWords(dictionary, wordOne, wordTwo)) {
+            //Output shortest word ladder from word 1 to word 2
+            getWordLadder(dictionary, wordOne, wordTwo);
+            cout << endl;
+        }
+    }
     cout << "Have a nice day." << endl;
     return 0;
 }
 
 /*
- * Function: nameImageFileToOpen
- * Usage: Prompts the user to open an image based on the file name. Adds the
- * image to the window if the image is found
- * Params: GBufferedImage, GWindow
+ * Function: getDictionary
+ * Usage: Prompts the user to get the reference dictionary by typing the file name. Reprompts the
+ * user if an invalid name is given.
+ * Params: dictionary (Lexicon)
  * -----------------------------------------------------------------
  * Returns: None. Void function
 */
 
-void getDictionary(Lexicon &dictionary){
-    string filename = promptUserForFile("Dictionary file name: ","Unable to open that file. Try again.");
+void getDictionary(Lexicon &dictionary) {
+    string filename = promptUserForFile(
+                "Dictionary file name: ","Unable to open that file. Try again.");
     dictionary = Lexicon(filename);
     }
 
-void getWords(const Lexicon &dictionary){
-    //TO DO
-    //Ask for word 1 "Word 1 (or hit Enter to quit): "
-    //Ask for word 2 "Word 2 (or hit Enter to quit): "
+/*
+ * Function: getWord
+ * Usage:
+ * Prompts user to input a word and strips the case and white spaces
+ * Params: prompt (string)
+ * -----------------------------------------------------------------
+ * Returns: word (string)
+ * NOTE for enhancement: this function would check whether the word is a valid dictionary word here
+ * instead and repromt the user to re-enter the word prior to checking other validations
+*/
+
+string getWord(const string &prompt) {
+    string word = getLine(prompt);
+     word = toLowerCase(trim(word));
+     return word;

```

Comment:
You can even return the modified word directly instead of reassigning it back to `word`.

```suggestion
    return toLowerCase(trim(word));

```